src/Nethermind/Nethermind.Serialization.Rlp/HeaderDecoder.cs

```diff
@@ -77,13 +77,19 @@ public class HeaderDecoder : IRlpValueDecoder<BlockHeader>, IRlpStreamDecoder<Bl
             {
                 blockHeader.WithdrawalsRoot = decoderContext.DecodeKeccak();
 
-                if (itemsRemaining == 3 && decoderContext.Position != headerCheck)
+                if (itemsRemaining >= 3 && decoderContext.Position != headerCheck)
                 {
                     blockHeader.DataGasUsed = decoderContext.DecodeULong(allowLeadingZeroBytes: false);
                     blockHeader.ExcessDataGas = decoderContext.DecodeULong(allowLeadingZeroBytes: false);
                 }
+
+                if (itemsRemaining >= 4 && decoderContext.Position != headerCheck)
```

Comment:
Don't we have spec based mechanism now for decoding? @deffrian @MarekM25 
If not we might want to potentially refactor and decode based on: block number and timestamp?

Reply:
we had a conversation about having the spec exposed to the decoders/encoders, and the conclusion was we keep them not coupled with ReleaseSpec (I kinda dont like it, cause I know at some point we wont have a workaround to avoid it)